Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -3524,21 +3524,29 @@ private async Task<PipExecutionStep> ExecutePipStep(RunnablePip runnablePip)
             }
         }
 
-        private static void FlagSharedOpaqueOutputs(IPipExecutionEnvironment environment, ExecutionResult executionResult)
+        private void FlagSharedOpaqueOutputs(IPipExecutionEnvironment environment, ProcessRunnablePip process)

```

Comment:
Here, in addition to `SharedDynamicDirectoryWriteAccesses` we apply the special timestamps to all outputs that are within some output directory